src/vaadin-time-picker.html

```diff
@@ -686,12 +686,15 @@
         }
 
         /**
-         * Returns true if `value` is valid, and sets the `invalid` flag appropriately.
+         * Validates the field and sets the `invalid` property based on the result.
          *
-         * @return {boolean} True if the value is valid and sets the `invalid` flag appropriately
+         * The method fires a `validated` event with the result of the validation.
          */
         validate() {
-          return !(this.invalid = !this.checkValidity());
+          const isValid = this.checkValidity();
+          this.invalid = !isValid;
+          this.dispatchEvent(new CustomEvent('validated', {detail: {valid: isValid}}));
+          return isValid;
```

Comment:
It seems we've also missed this JSDoc part in this and previous PRs:

https://github.com/vaadin/web-components/pull/4084/files#diff-ce25fa3626d29f442b1b98935ad7c0909865efc4375789d17e1f7b3b18fe392dR34-R40

It has to be added for the event to show up in the API documentation.